win32/Lib/win32serviceutil.py

```diff
@@ -481,8 +481,12 @@ def GetServiceClassString(cls, argv = None):
             path = os.path.split(fname)[0]
             # Eaaaahhhh - sometimes this will be a short filename, which causes
             # problems with 1.5.1 and the silly filename case rule.
-            # Get the long name
-            fname = os.path.join(path, win32api.FindFiles(fname)[0][8])
+            filelist = win32api.FindFiles(fname)
```

Comment:
What does argv[0] look like in this case? I'm surprised GetFullPathName works while FindFiles does not.

Reply:
The short answer is "MADST_Client_Service.exe". At least, this is what's returned when running from windows command prompt. When run from mingw, argv[0] is the full path to the same file and the bug does not occur. Implementation for this is at https://github.com/graysonhead/MADST/blob/master/client/MADST_Client_Service.py . GetFullPathName returns the expected result. The exe is generated using pyinstaller as defined in build.bat in the same directory of that repo.